pkgs/applications/networking/mailreaders/thunderbird/default.nix

```diff
@@ -1,196 +1,221 @@
-        --set MOZ_LEGACY_PROFILES 1
-        --set MOZ_ALLOW_DOWNGRADE 1
-        # https://github.com/NixOS/nixpkgs/pull/61980
-        --set SNAP_NAME "thunderbird"
-      )
-      ${
-        # We wrap manually because wrapGAppsHook does not detect the symlink
-        # To mimic wrapGAppsHook, we run it with dontWrapGApps, so
-        # gappsWrapperArgs gets defined correctly
-        lib.optionalString enableGTK3 "wrapGAppsHook"
-      }
-
-      # "$binary" is a symlink, replace it by the wrapper
-      rm "$binary"
-      makeWrapper "$target" "$binary" "''${gappsWrapperArgs[@]}"
-
-      ${ let desktopItem = makeDesktopItem {
-          name = "thunderbird";
-          exec = "thunderbird %U";
-          desktopName = "Thunderbird";
-          icon = "$out/lib/thunderbird/chrome/icons/default/default256.png";
-          genericName = "Mail Reader";
-          categories = "Application;Network";
-          mimeType = stdenv.lib.concatStringsSep ";" [
-            # Email
-            "x-scheme-handler/mailto"
-            "message/rfc822"
-            # Newsgroup
-            "x-scheme-handler/news"
-            "x-scheme-handler/snews"
-            "x-scheme-handler/nntp"
-            # Feed
-            "x-scheme-handler/feed"
-            "application/rss+xml"
-            "application/x-extension-rss"
-          ];
-        }; in desktopItem.buildCommand
-      }
-    '';
-
-  postFixup =
-    # Fix notifications. LibXUL uses dlopen for this, unfortunately; see #18712.
-    ''
-      patchelf --set-rpath "${lib.getLib libnotify
-        }/lib:$(patchelf --print-rpath "$out"/lib/thunderbird*/libxul.so)" \
-          "$out"/lib/thunderbird*/libxul.so
-    '';
+  postConfigure = ''
+    cd obj-*
+  '';
+
+  makeFlags= lib.optionals enableOfficialBranding [
+    "MOZILLA_OFFICIAL=1"
+    "BUILD_OFFICIAL=1"
+  ];
+
+  doCheck = false;
+  dontWrapGApps = true; # manually done
+  postInstall = ''
+    # TODO: Move to a dev output?
+    rm -rf $out/include $out/lib/thunderbird-devel-* $out/share/idl
+
+    # $binary is a symlink to $target.
+    # We wrap $target by replacing the $binary symlink.
+    local target="$out/lib/thunderbird/thunderbird"
+    local binary="$out/bin/thunderbird"
+
+    # Wrap correctly, this is needed to
+    # 1) find Mozilla runtime, because argv0 must be the real thing,
+    #    or a symlink thereto. It cannot be the wrapper itself
+    # 2) detect itself as the default mailreader across builds
+    gappsWrapperArgs+=(
+      --argv0 "$target"
+      --set MOZ_APP_LAUNCHER thunderbird
+      --set MOZ_LEGACY_PROFILES 1
+      --set MOZ_ALLOW_DOWNGRADE 1
+      # https://github.com/NixOS/nixpkgs/pull/61980
+      --set SNAP_NAME "thunderbird"
+    )
+    ${
+      # We wrap manually because wrapGAppsHook does not detect the symlink
+      # To mimic wrapGAppsHook, we run it with dontWrapGApps, so
+      # gappsWrapperArgs gets defined correctly
+      lib.optionalString gtk3Support "wrapGAppsHook"
+    }
+
```

Comment:
Do you know when this diff was added? I believe this bug has been fixed.

Reply:
I do not, but I can give it a shot and see if it works; doing that now.

Reply:
Done!